packages/core-typings/src/IMethodThisType.ts

```diff
@@ -1,16 +1,18 @@
+import type { ISocketConnection } from './ISocketConnection';
+
 export interface IMethodConnection {
-	id: string;
-	close: Function;
-	onClose: Function;
-	clientAddress: string;
-	httpHeaders: Record<string, any>;
+  id: string;
+  close(fn: (...args: any[]) => void): void;
+  onClose(fn: (...args: any[]) => void): void;
+  clientAddress: string;
+  httpHeaders: Record<string, any>;
 }
 
 export interface IMethodThisType {
-	isSimulation: boolean;
-	userId: string | null;
-	connection: IMethodConnection | null;
-	setUserId(userId: string): void;
-	unblock(): void;
-	twoFactorChecked: boolean | undefined;
+  isSimulation: boolean;
+  userId: string | null;
+  connection: ISocketConnection | null;
```

Comment:
Same here, why do we need to switch to ISocketConnection? I'm not seeing any changes to existing code that are using these types (as in IMethodThisType).

Reply:
Everything about socket events expects connection as ISocketConnection

[apps/meteor/server/services/sauMonitor/events.ts](https://github.com/RocketChat/Rocket.Chat/blob/chore/session-login-token/apps/meteor/server/services/sauMonitor/events.ts#L5)